cmd/crates/soroban-spec-tools/src/contract.rs

```diff
@@ -276,3 +285,214 @@ fn format_name(lib: &StringM<80>, name: &StringM<60>) -> String {
         name.to_utf8_string_lossy()
     }
 }
+
+pub fn pretty_spec(spec: &Vec<ScSpecEntry>) -> String {
```

Comment:
I think we should use the implementation that is available at the following link to generate the interface, because it generates verified valid rust code that gets used in contracts:
https://docs.rs/soroban-spec-rust

I apologise for not pointing out the existence of that crate earlier as it would have saved you some time.

The existing `generate` functions located in that crate will generate code that looks like this:
```
$ stellar contract bindings rust --wasm test_empty.wasm
pub const WASM: &[u8] = soroban_sdk::contractfile!(
    file = "../rs-soroban-sdk/target/wasm32-unknown-unknown/release/test_empty.wasm",
    sha256 = "ec784e29ec4b9dcceb014302805d6a85485b71b06438f6ce69e6baafc18997ff"
);
#[soroban_sdk::contractclient(name = "Client")]
pub trait Contract {
    fn empty(env: soroban_sdk::Env);
}
```

That output is currently available on this command:
```
stellar contract bindings rust
```

So I guess, there's some overlap between the `contract info interface` command and this command. However, the purposes of the two commands are a little different.

The `bindings generate`'s goal is to generate client code to use for calling a contract, which is why the generated code includes the `contractclient` attribute, and the `WASM` bytes.

The `contract info interface`'s goal is to really just report the interface itself.

So I think what we should do is in this PR you could move forward with generating the same output as `bindings generate`, and then as a next step we can make a change to `soroban-spec-rust` to expose a function for generating only the interface. It'll just require a small refactor of the crate so that we can get at the interface without the other noise.

Reply:
I've opened a PR on the `soroban-spec-rust` crate adding a function to get at just the interface:
- https://github.com/stellar/rs-soroban-sdk/pull/1311

Reply:
Oh got it sorry didn't do enough DD before spending some time on the prettyfying it. I'll use the new function from soroban-spec-rust crate for pretty printing.

Reply:
Left one comment in the PR, replaced the usage of this method with `generate_without_file`